app/src/main/scala/App.scala

```diff
@@ -1,10 +1,7 @@
 import pureconfig._
-import pureconfig.generic.derivation.default._
-
-object App {
-  implicit def personReader: ConfigReader[Person] = ConfigReader.derived[Person]
 
+object App extends AppVersionSpecific {
   def main(args: Array[String]): Unit = {
-    println(ConfigSource.string("{ name = foo }").load[Person])
+    println(ConfigSource.string("{ name = foo, age = 1 }").load[Person])
```

Comment:
This has to pass without `age` added here. The point is that we can make a change in a library without breaking existing code, binary- or runtime. If users have to change _anything_ when they upgrade to avoid a crash, then it's not compatible.

Reply:
I've brought back the default arguments (I don't know why I removed them in the first place).

Still works on everything (`++2.13.* appOld/run`, `++2.13.* appNew/run`, `++3.* appOld/run`). The only exception is `++3.* appNew/run`. That's a [bug in Scala 3 PureConfig](https://github.com/pureconfig/pureconfig/issues/1488), it doesn't work with default arguments, so it can't parse it. But it _doesn't_ crash, it just fails parsing.